File: codeforces/800/longWord.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n;
let count = 0;
const words = [];

rl.on("line", (input) => {
  if (!n) {
    n = parseInt(input);
  } else {
    const word = input.trim();
    if (word.length > 10) {
      words.push(word[0] + (word.length - 2) + word[word.length - 1]);
    } else {
      words.push(word);
    }
    count++;
    if (count === n) {
      console.log(words.join("\n"));
      rl.close();
    }
  }
});

File: codeforces/800/longWord.rs
use std::io;

fn main() {
    let mut input = String::new();

    // Read the number of words
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let n: usize = input.trim().parse().expect("Invalid number of words");

    // Process each word
    for _ in 0..n {
        input.clear();
        io::stdin().read_line(&mut input).expect("Failed to read input");
        let word = input.trim();

        // Abbreviate the word if it's too long
        if word.len() > 10 {
            let abbreviation = format!(
                "{}{}{}",
                word.chars().next().unwrap(),
                word.len() - 2,
                word.chars().last().unwrap()
            );
            println!("{}", abbreviation);
        } else {
            println!("{}", word);
        }
    }
}
